core/app/models/spree/order.rb

```diff
@@ -596,8 +596,9 @@ def add_store_credit_payments
 
       if matching_store_credits.any?
         payment_method = Spree::PaymentMethod::StoreCredit.first
+        sorter = Spree::Config.order_credit_sorter_class.new(self)
```

Comment:
What about passing store credits to sort in the initializer instead of the order? Having the sorter know about the order is something that only applies in your domain, while the store credits will be needed in any scenario.

Reply:
I'm definitely fine with passing the credits in the constructor! Will update after we're aligned on the other comment, though.